components/eam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -1184,19 +1191,24 @@ subroutine micro_p3_tend(state, ptend, dtime, pbuf)
     !================
     !Since the host model needs wet mixing ratio tendencies(state vector has wet mixing ratios),
     !we need to convert dry mixing ratios from P3 to wet mixing ratios before extracting tendencies
-    !NOTE: water vapor mixing ratio argument in calculate_wetmmr_from_drymmr function has to be dry water vapor mixing ratio
-
-    qv_wet_out(:ncol,:pver) = calculate_wetmmr_from_drymmr(ncol, pver, qv_dry,  qv_dry)
-    cldliq(:ncol,:pver)     = calculate_wetmmr_from_drymmr(ncol, pver, cldliq,  qv_dry)
-    numliq(:ncol,:pver)     = calculate_wetmmr_from_drymmr(ncol, pver, numliq,  qv_dry)
-    rain(:ncol,:pver)       = calculate_wetmmr_from_drymmr(ncol, pver, rain,    qv_dry)
-    numrain(:ncol,:pver)    = calculate_wetmmr_from_drymmr(ncol, pver, numrain, qv_dry)
-    ice(:ncol,:pver)        = calculate_wetmmr_from_drymmr(ncol, pver, ice,     qv_dry)
-    numice(:ncol,:pver)     = calculate_wetmmr_from_drymmr(ncol, pver, numice,  qv_dry)
-    qm(:ncol,:pver)         = calculate_wetmmr_from_drymmr(ncol, pver, qm,      qv_dry)
-    rimvol(:ncol,:pver)     = calculate_wetmmr_from_drymmr(ncol, pver, rimvol,  qv_dry)
-
-    temp(:ncol,:pver) = th(:ncol,:pver)/inv_exner(:ncol,:pver)
+
+    ratio_local(:ncol,:pver) = state%pdeldry(:ncol,:pver)/state%pdel(:ncol,:pver)
+
+    qv_wet_out (:ncol,:pver) = qv_dry (:ncol,:pver) *ratio_local(:ncol,:pver)
+    cldliq     (:ncol,:pver) = cldliq (:ncol,:pver) *ratio_local(:ncol,:pver)
+    numliq     (:ncol,:pver) = numliq (:ncol,:pver) *ratio_local(:ncol,:pver)
+    rain       (:ncol,:pver) = rain   (:ncol,:pver) *ratio_local(:ncol,:pver)
+    numrain    (:ncol,:pver) = numrain(:ncol,:pver) *ratio_local(:ncol,:pver)
+    ice        (:ncol,:pver) = ice    (:ncol,:pver) *ratio_local(:ncol,:pver)
+    qm         (:ncol,:pver) = qm     (:ncol,:pver) *ratio_local(:ncol,:pver)
+    numice     (:ncol,:pver) = numice (:ncol,:pver) *ratio_local(:ncol,:pver)
+    rimvol     (:ncol,:pver) = rimvol (:ncol,:pver) *ratio_local(:ncol,:pver)
+
+    !new - old
```

Comment:
It isn't clear to me why dT would need wet/dry conversion since I don't think of temperature as being wet or dry or even having units of "per kg air". Could you add a comment to the code providing hints about what you're doing here?

Reply:
Thanks, i added a comment.